core/src/Kokkos_Macros.hpp

```diff
@@ -374,11 +374,11 @@
 #endif
 
 #if !defined(KOKKOS_INLINE_FUNCTION_DELETED)
-#define KOKKOS_INLINE_FUNCTION_DELETED inline
+#define KOKKOS_INLINE_FUNCTION_DELETED
 #endif
 
 #if !defined(KOKKOS_DEFAULTED_FUNCTION)
-#define KOKKOS_DEFAULTED_FUNCTION inline
+#define KOKKOS_DEFAULTED_FUNCTION
```

Comment:
I agree with removing the inline, so that also leads me to ask:  if we do that, it becomes an empty macro, so what is the point of having it at all? just in case we want to redefine to something else later?

Reply:
Application codes use at least `KOKKOS_DEFAULTED_FUNCTION` so I don't think we should remove it right away but we could think about deprecating it in some way (or don't in case there will be a need to do something here).

Reply:
but do application codes use that macro also for free functions or just for class member fncs?

Reply:
Free functions cannot be `default` anyway, right? I'm not quite sure how much `delete` is used, though.

Reply:
yes, agree! 
In any case my main point was that if we make a macro empty, I think we should deprecate it at some point, since I don't quite see the purpose of an empty macro